src/pivot-table/components/cells/DimensionValue/utils/get-style.ts

```diff
@@ -0,0 +1,104 @@
+import { getHoverColor as getContrastingColor } from "@qlik/nebula-table-utils/lib/utils";
+import type { Cell, StyleService } from "../../../../../types/types";
+import { BOLD_FONT_WEIGHT } from "../../../../constants";
+
+type GetStyleServiceValueProps = {
+  cell: Cell;
+  styleService: StyleService;
+};
+
+type GetBackgroundColorProps = GetStyleServiceValueProps & {
+  isCellSelected: boolean;
+};
+
+type GetBackgroundImageProps = {
+  cell: Cell;
+  backgroundColor?: string;
+  isCellLocked: boolean;
+};
+
+type GetColorProps = GetStyleServiceValueProps & { isCellSelected: boolean };
+
+enum Properties {
+  Background = "background",
+  Color = "color",
+  FontWeight = "fontWeight",
+  FontStyle = "fontStyle",
+  TextDecoration = "textDecoration",
+}
+
+const lockedColorModifiers = { brighter: 0.5, darker: 0.3, opacity: 0.03 };
+
+const getLockedBackground = (originalBackgroundColor: string) => {
+  const strip = getContrastingColor(originalBackgroundColor, lockedColorModifiers);
+
+  return `repeating-linear-gradient(
+    -45deg,
+    ${strip} 0px 2px,
+    ${originalBackgroundColor} 0px 4px
+  )`;
+};
+
+const getStyleServiceValue = (prop: Properties, { cell, styleService }: GetStyleServiceValueProps) => {
+  if (prop === Properties.Background || prop === Properties.Color) {
+    const expressionValue = cell.expressionColor[prop];
+    if (expressionValue) {
+      return expressionValue;
+    }
+  }
+
+  if (cell.isNull) {
+    return styleService.nullValues[prop];
+  }
+
+  if (cell.isPseudoDimension) {
+    return styleService.measureLabels[prop];
+  }
+
+  if (cell.isTotal) {
+    return styleService.totalValues[prop];
+  }
+
+  return styleService.dimensionValues[prop];
+};
+
+export const getBackgroundColor = ({ styleService, isCellSelected, cell }: GetBackgroundColorProps) => {
+  if (isCellSelected) {
+    return "#0aaf54";
+  }
+
+  return getStyleServiceValue(Properties.Background, { cell, styleService });
+};
+
+export const getBackgroundImage = ({ backgroundColor, isCellLocked, cell }: GetBackgroundImageProps) => {
+  if (isCellLocked && backgroundColor && !cell.isPseudoDimension && !cell.isEmpty && !cell.isNull && !cell.isTotal) {
+    return getLockedBackground(backgroundColor);
```

Comment:
I think you forgot to include this fix: https://github.com/qlik-oss/sn-pivot-table/pull/433
probably you branched off before that fix

Reply:
No this fix is included. See `backgroundClip` in `Container.tsx`

Reply:
the magic here was to add background clip in `repeating-linear-gradient(...)` specifically and not in the container element since container is not responsible to show the striped bg effect. 
<img width="1924" alt="Screenshot 2023-10-26 at 13 11 20" src="https://github.com/qlik-oss/sn-pivot-table/assets/29652890/b5c51205-b942-429a-9188-04ed8280b9be">


Current behaviour with your changes is:

https://github.com/qlik-oss/sn-pivot-table/assets/29652890/f55f2698-4f08-49ab-a6a3-43d19d7effd5

Reply:
I've change it now so that `getLockedBackground` returns a background with `border-box`. Could you take a second look @a-m-dev ?